php-frontend/src/main/java/org/sonar/php/tree/impl/declaration/ClassDeclarationTreeImpl.java

```diff
@@ -0,0 +1,207 @@
+    this.openCurlyBraceToken = openCurlyBraceToken;
+    this.members = members;
+    this.closeCurlyBraceToken = closeCurlyBraceToken;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken modifierToken() {
+    return modifierToken;
+  }
+
+  @Override
+  public SyntaxToken classEntryTypeToken() {
+    return classEntryTypeToken;
+  }
+
+  @Override
+  public IdentifierTree name() {
+    return name;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken extendsToken() {
+    return extendsToken;
+  }
+
+  @Nullable
+  @Override
+  public NamespaceNameTree superClass() {
+    return superClass;
+  }
+
+  @Nullable
+  @Override
+  public SyntaxToken implementsToken() {
+    return implementsToken;
+  }
+
+  @Override
+  public SeparatedList<NamespaceNameTree> superInterfaces() {
+    return superInterfaces;
+  }
+
+  @Override
+  public SyntaxToken openCurlyBraceToken() {
+    return openCurlyBraceToken;
+  }
+
+  @Override
+  public List<ClassMemberTree> members() {
+    return members;
+  }
+
+  @Override
+  public SyntaxToken closeCurlyBraceToken() {
+    return closeCurlyBraceToken;
+  }
+
+  @Override
+  public Kind getKind() {
+    return KIND;
+  }
+
+  @Override
+  public Iterator<Tree> childrenIterator() {
+    return Iterators.concat(
+        Iterators.forArray(modifierToken, classEntryTypeToken, name, extendsToken, superClass, implementsToken),
+        superInterfaces.elementsAndSeparators(Functions.<NamespaceNameTree>identity()),
+        Iterators.singletonIterator(openCurlyBraceToken),
+        members.iterator(),
+        Iterators.singletonIterator(closeCurlyBraceToken)
+    );
+  }
+
+  @Override
+  public void accept(TreeVisitor visitor) {
+    visitor.visitClassDeclaration(this);
+  }
+
+  public static ClassDeclarationTree createInterface(
+      InternalSyntaxToken interfaceToken, IdentifierTree name,
+      @Nullable InternalSyntaxToken extendsToken, SeparatedList<NamespaceNameTree> interfaceList,
+      InternalSyntaxToken openCurlyBraceToken, List<ClassMemberTree> members, InternalSyntaxToken closeCurlyBraceToken
+  ) {
+    return new ClassDeclarationTreeImpl(
+        Kind.INTERFACE_DECLARATION,
+        null,
+        interfaceToken,
+        name,
+        extendsToken,
+        null,
+        null,
+        interfaceList,
+        openCurlyBraceToken,
+        members,
+        closeCurlyBraceToken
+    );
+  }
+
+  public static ClassDeclarationTree createTrait(
+      InternalSyntaxToken traitToken, IdentifierTree name,
+      InternalSyntaxToken openCurlyBraceToken, List<ClassMemberTree> members, InternalSyntaxToken closeCurlyBraceToken
+  ) {
+    return new ClassDeclarationTreeImpl(
+        Kind.TRAIT_DECLARATION,
+        null,
+        traitToken,
+        name,
+        null,
+        null,
+        null,
+        SeparatedList.<NamespaceNameTree>empty(),
+        openCurlyBraceToken,
+        members,
+        closeCurlyBraceToken
+    );
+  }
+
+  public static ClassDeclarationTree createClass(
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 10 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00107)